src/app/courses/course-item/course-item.component.ts

```diff
@@ -28,46 +27,15 @@ implements OnInit,OnChanges, AfterContentInit, AfterContentChecked, AfterViewIni
   faTrash: IconDefinition = faTrash;
   faCalendarAlt: IconDefinition = faCalendarAlt;
   faClock: IconDefinition = faClock;
+  faStar: IconDefinition = faStar;
 
   constructor() { }
 
-  ngOnChanges()	{
-    console.log('ngOnChanges');
-  }
-
-  ngOnInit() {
-    console.log('ngOnInit');
-  }
-
-  // ngDoCheck() {
-  //   console.log('ngOnInit');
-  // }
-
-  ngAfterContentInit(): void {
-    console.log('ngAfterContentInit');
-  }
-
-  ngAfterContentChecked(): void {
-    console.log('ngAfterContentChecked');
-  }
-
-  ngAfterViewInit(): void {
-    console.log('ngAfterViewInit');
-  }
-
-  ngAfterViewChecked(): void {
-    console.log('ngAfterViewChecked');
-  }
-
-  ngOnDestroy(): void {
-    console.log('ngOnDestroy');
-  }
-
   onEdit() {
     this.changed.emit(this.data);
   }
 
   onDelete() {
-    this.deleted.emit(this.data.id);
+    this.deleted.emit(this.data && this.data.id);
```

Comment:
I believe we don't need to emit data in case it absent

Reply:
done